### DISCUSSION TITLE
BerkeleyDB: How to dump in a turtle file

### DISCUSSION THREAD

**Discussion by bronger** [RDFLib/rdflib#2006] [Q&A]
I have created a Berkeley DB store with all triples. How can I create a turtle file from that using rdflib? If I open it with

    graph = rdflib.Graph("BerkeleyDB")
    graph.open("/path/to/db")

then `graph` is empty. Thank you!

**Comment by ghost** [comment]
There's [a fully-worked example](https://rdflib.readthedocs.io/en/stable/_modules/examples/berkeleydb_example.html#example_1), basically the `create` kwarg for BerkeleyDB's `open` method defaults to `create=True` which has the unfortunate effect of overwriting any existing BerkeleyDB structure with an as-yet-unpopulated new structure.

If you use `graph.open("/path/to/db", create=False)`, you should be good.

To serialize as turtle: `graph.serialize("/tmp/graph.ttl", format="turtle")` (or provide a handle to an open file ... see the [API docs](https://rdflib.readthedocs.io/en/stable/apidocs/rdflib.html#rdflib.graph.Graph.serialize) for full details.)
